### DISCUSSION TITLE
There is no OFFLINE MODE

### DISCUSSION THREAD

**Discussion by mike7ang1rdz** [microsoft/aspire#2492] [General]
I have a asp.net core app that runs as a service on the user machine.

  It would be nice if remote requests can be saved locally and then synchronize when online,and complete the response.

**Comment by davidfowl** [comment]
Can you explain more? What's the scenario?

**Comment by mike7ang1rdz** [comment]
Basically we have a dozen of services running on azure. But one of them runs on the user's machine (remote areas without internet) so they process everything offline (save in SQLite) and then synchronize using queues/messages.  Everything would be easier if all this offline mechanism. Could be handled natively.

**Comment by ReubenBond** [comment]
Would you mind elaborating on what you would like an "offline mode" in Aspire to do? Is this a scenario where you have a desktop/mobile application which has intermittent connectivity to the cloud and you want the system to operate regardless? My initial reaction, if my understanding is accurate, is that it is out of scope for Aspire in it's current form and you would need to handle that in your application code. Local-first/offline-first apps are the realm of CRDTs (eg, see Automerge https://automerge.org/blog/2023/11/06/automerge-repo/).

**Comment by mike7ang1rdz** [comment]
Thanks @ReubenBond that's correct.  It is doing that approach already.  If Aspire implements that natively would be awesome. 👌
